src/Microsoft.Performance.SDK/Processing/DefaultProcessingMode.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+
+namespace Microsoft.Performance.SDK.Processing
+{
+    /// <summary>
+    ///     A default <see cref="IProcessingMode"/>.
+    /// </summary>
+    public class DefaultProcessingMode
+        : IProcessingMode
+    {
```

Comment:
Lets add a static getter here :

`public static readonly DefaultProcessingMode Default = new DefaultProcessingMode();`

Reply:
I'd rather let plugins create their own `IProcessingMode` classes so that they can store extra information on the class if they need it. For instance, a plugin might want to store some metadata about _how_ a processing mode should be configured on the class.

With the suggest approach, that wouldn't be possible. Plugins would also have to check for the processing mode to use with Guid comparisons (i.e. `if (group.ProcessingMode.Guid == MyProcessingModeGuid)`) instead of a type check (i.e. `if (group.ProcessingMode is MyProcessingMode)`).

I am going to seal this class though.

Reply:
I do not see the benefit of adding a `public static readonly DefaultProcessingMode Default` field. When/why would it be used?